Kingfisher/ImageCache.swift

```diff
@@ -608,11 +604,11 @@ public extension ImageCache {
                     }
             }
             
-            dispatch_async(dispatch_get_main_queue(), { () -> Void in
-                if let completionHandler = completionHandler {
+            if let completionHandler = completionHandler {
+                dispatch_async(dispatch_get_main_queue(), { () -> Void in
                     completionHandler(size: diskCacheSize)
-                }
-            })
```

Comment:
I guess we could just simplify the API from an optional `completionHandler` to a non-option one. There would be meaningless if no callback supplied. Since that, we could get rid of the optional chaining here. Would you please file another commit for it? Thanks!